### PR TITLE
Remove dependency on identityclient.DecodeAccessToken from hpegl-provider-lib

### PR DESCRIPTION
**Pull request by thezelk**
1. **DecodeAccessToken** copied from iam-lib and pasted in token-util package in hpegl-provider-lib.
2. **iam-lib dependency** removed in serviceclient package (handler.go).
3. **logrusv1.8.1** added in go.mod as parseJWT function in token-utils needs this dependency.
4. **Tests from iam-lib** copied/pasted to token-utils_test.go file.

### CODE REVIEW

File: pkg/token/token-util/token-util_test.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+			},
+			want: Token{
+				Issuer:          "https://idp-broker.dev.hpedevops.net/auth/realms/master",
+				Subject:         "users/10ac21d4-589b-43ca-a434-ad1999f3950a",
+				TenantID:        "8r40-ld9p",
+				Expiry:          1557962764,
+				Type:            "Bearer",
+				IssuedAt:        1557948364,
+				Nonce:           "12345",
+				AuthorizedParty: "onesphere-auth",
```
Comment (howleyt) [inline pkg/token/token-util/token-util_test.go:56]: The onesphere reference is a bit strange here. Maybe change if the string doesn't matter.

File: pkg/token/token-util/token-util_test.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+					"HRwczovL2hwZS1ncmVlbmxha2Uub2t0YXByZXZpZXcuY29tL29hdXRoMi9kZWZhdWx0IiwiYXVkIjoiYXBpOi8vZGVmYXVsdC" +
+					"IsImlhdCI6MTU2NjQwMzk1MCwiZXhwIjoxNTY2NDA3NTUwLCJjaWQiOiIwb2FtczQ2OGZ3eHhZcDZ6UjBoNyIsInVpZCI6IjA" +
+					"wdW4xem9xcHFSRDEwWDVLMGg3Iiwic2NwIjpbIm9wZW5pZCIsInByb2ZpbGUiLCJlbWFpbCJdLCJzdWIiOiJyeWFuLmJyYW5k" +
+					"dEBocGUuY29tIiwidGVuYW50SWQiOiJocGUtZ3JlZW5sYWtlLWludGcifQ.dwo2de1BSam5kME8bqQnQWBys6fGfRfGR05z1l" +
+					"fJSq1BTIuXGPcxpkmKOyjOFzPR1VoteQp1ZwC02Qj9WyYabgGI6LZ5odPv01GT9hAVUc0RijFPrp9w666LDPv_1LAyzgeccfAl" +
+					"5MZNeVDk7OdAZX7UiSj_mLnDAJv9Dzln-CMJA3JgA5GjZVZo0mdvfzD6mCqNL594za_ouZt5_Pp4TPoAAzGkYSjyv8LJWT0Zrs" +
+					"ruzPKlS1UTJR6wUSxlqCSQnjZtXQtE5UugQmxdV1F_SC6HxikabnCu-6zLTPgJANUQ1XXKSE2brEgkYIx4suTv6IzejGz1Ccnw" +
+					"O__RLEJ2gQ",
+			},
+			want: Token{
```
[the remark this answers is not in the document]
Comment (eamonnotoole) [reply pkg/token/token-util/token-util_test.go:74]: I asked Joe Keen about that and he said that it's fine to make the code open.

**Comment by stuart-mclaren-hpe** [comment]
> logrusv1.8.1 added in go.mod as parseJWT function in token-utils needs this dependency.

We should think about what logging library we want to use. It feels like we're importing logrus just to make the code compile.

Are folks ok with adopting logrus for logging for hpegl-provider-lib?

**Review by stuart-mclaren-hpe — COMMENTED** [review]
I think it would be good to remove the logrus dependency in favour of integrating with what terraform usually do:

https://www.terraform.io/docs/extend/debugging.html

eg

```
import log
log.Printf("[DEBUG] Hello")
```

**Review by stuart-mclaren-hpe — CHANGES_REQUESTED** [review]
It would be good to remove the logrus dependency.

File: pkg/token/token-util/token-util_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (howleyt) [reply pkg/token/token-util/token-util_test.go:74]: Thanks, understood.

File: pkg/token/token-util/token-util_test.go (hunk 1 above)
Comment (howleyt) [reply pkg/token/token-util/token-util_test.go:56]: Can ignore, not critical.

File: pkg/token/token-util/token-util.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		return Token{}, fmt.Errorf("oidc: malformed jwt: %v", err)
+	}
+	var token Token
+	if err := json.Unmarshal(payload, &token); err != nil {
+		log.Fatal(fmt.Sprintf("oidc: failed to unmarshal claims: %v", err))
+		return Token{}, fmt.Errorf("oidc: failed to unmarshal claims: %v", err)
+	}
+
+	if token.UserID != "" {
+		// Okta User token
```
Comment (stuart-mclaren-hpe) [inline pkg/token/token-util/token-util.go:53]: I think this reference to okta should be removed.

Comment (thezelk) [reply pkg/token/token-util/token-util.go]: The reference to Okta is now removed.

File: pkg/token/token-util/token-util.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
+//(C) Copyright 2019 Hewlett Packard Enterprise Development LP
```
[the remark this answers is not in the document]
Comment (thezelk) [reply pkg/token/token-util/token-util.go]: Copyright updated as requested.

**Review by stuart-mclaren-hpe — APPROVED** [review]

**Pull request merged**